store/mongo/datastore_mongo.go

```diff
@@ -406,9 +419,19 @@ func (db *DataStoreMongo) DeleteToken(ctx context.Context, jti oid.ObjectID) err
 }
 
 func (db *DataStoreMongo) DeleteTokens(ctx context.Context) error {
-	c := db.client.Database(ctxstore.DbFromContext(ctx, DbName)).
+	c := db.client.Database(DbName).
 		Collection(DbTokensColl)
-	_, err := c.DeleteMany(ctx, bson.D{})
+
+	id := identity.FromContext(ctx)
+	filter := bson.M{}
+	// I have some doubts in here: we are dropping all the tokens if
+	// tenant claim in the JWT is empty
+	if id != nil {
+		if id.Tenant != "" {
+			filter = bson.M{dbFieldTenantClaim: id.Tenant}
+		}
```

Comment:
(Addressing your comment): Default to filtering by `{mender.tenant: null}` which should preserve the old behavior.
```suggestion
	filter := bson.M{dbFieldTenantClaim: nil}
	if id != nil && id.Tenant != "" {
		filter[dbFieldTenantClaim] = id.Tenant
```

Reply:
thanks lovely.